src/Avalonia.Base/Media/TextFormatting/TextLine.cs

```diff
@@ -218,24 +218,14 @@ public abstract class TextLine
                         return Math.Max(0, (paragraphWidth - width) / 2);
 
                     case TextAlignment.Right:
-                        return Math.Max(0, paragraphWidth - widthIncludingTrailingWhitespace);
+                        return flowDirection == FlowDirection.LeftToRight ? Math.Max(0, paragraphWidth - widthIncludingTrailingWhitespace) : 0;
```

Comment:
I think this `switch` was supposed to be moved outside the `if`? (i.e. removing the `if`)

Reply:
Alignment is inverted for RightToLeft

Reply:
Right, but this switch is inside an `if (flowDirection == FlowDirection.LeftToRight)`, and now it's handled in the ternary, I think you forgot to remove it?

Reply:
I will just invert the Alignment value to make it more readable and more compact